src/config.rs

```diff
@@ -147,29 +146,20 @@ impl Config {
             config.excerpt_separator = excerpt_separator.to_owned();
         };
 
-        #[cfg(all(feature="syntax-highlight"))]
-        {
-            use error::{Error, ErrorKind};
-
-            if let Some(theme) = yaml["syntax-highlight"]["theme"].as_str() {
-                match has_syntax_theme(theme) {
-                    Ok(true) => config.syntax_theme = theme.to_owned(),
-                    Ok(false) => {
-                        info!(concat!("Theme named '{}' is not found.",
-                                      "  Using default 'base16-ocean.dark'.",
-                                      "  Run `cobalt list-syntax-themes` to see available themes."),
-                              theme)
-                    }
-                    Err(Error(ErrorKind::UnsupportedPlatform(..), _)) => {
-                        info!(concat!("Theme named '{}' ignored.",
-                                      "  This build of cobalt does not include support",
-                                      " for syntax highlighting."),
-                              theme)
-                    }
-                    Err(_) => unreachable!(),
+        if let Some(theme) = yaml["syntax-highlight"]["theme"].as_str() {
+            let result: Result<()> = match has_syntax_theme(theme) {
+                Ok(true) => {
+                    config.syntax_theme = theme.to_owned();
+                    Ok(())
+                }
+                Ok(false) => Err(format!("Theme '{}' is unsupported", theme).into()),
+                Err(err) => {
+                    warn!("Theme named '{}' ignored. Reason: {:?}", theme, err);
+                    Ok(())
                 }
             };
-        }
+            result?;
+        };
```

Comment:
With `feature=syntax-highlight`, an invalid `syntax-highlight.theme` entry in `.cobalt.yml` is a fatal error.  This is intentional?

I had this fall back to a sane default.  Thought it might make for a better user experience, but I can see the advantage to the fatal error, especially for automated deployments.